pyiron_base/jobs/master/generic.py

```diff
@@ -263,6 +309,47 @@ def move_to(self, project):
                 child.move_to(project.open(self.job_name + "_hdf5"))
         super(GenericMaster, self).move_to(project)
 
+    def collect_output(self):
+        """Collect the generic output files of the individual jobs"""
+        if self.ref_job.server.run_mode.interactive:
+            output = self.ref_job["output/generic"]
+            if output is not None:
+                with self.project_hdf5.open("output/generic") as hh:
+                    for node in self.ref_job["output/generic"].list_nodes():
+                        if self.ref_job["output/generic/{}".format(node)] is not None:
+                            hh[node] = self.ref_job["output/generic/{}".format(node)]
+        ham_lst = [self.project_hdf5.inspect(child_id) for child_id in self.child_ids]
+        if (
+            "output" in ham_lst[0].list_groups()
+            and "generic" in ham_lst[0]["output"].list_groups()
+        ):
+            nodes = ham_lst[0]["output/generic"].list_nodes()
+            with self.project_hdf5.open("output/generic") as hh:
+                for node in nodes:
+                    if ham_lst[0]["output/generic/{}".format(node)] is not None:
+                        hh[node] = np.concatenate(
+                            [ham["output/generic/{}".format(node)] for ham in ham_lst],
+                            axis=0,
+                        )
+                    else:
+                        hh[node] = None
```

Comment:
This makes no sense for parallel masters

Reply:
What exactly doesn't make sense?

Reply:
It takes the output of only the first child job

Reply:
No, you only iterate over the groups of the first job `nodes = ham_lst[0]["output/generic"].list_nodes()`

Reply:
Yes, I don't think the nodes change for different child jobs, especially for `ParallelMaster`, so it doesn't matter which child gives the nodes.

Reply:
> Yes, I don't think the nodes change for different child jobs, especially for `ParallelMaster`, so it doesn't matter which child gives the nodes.

No, for the `ParallelMaster` you want to iterate over all children.

Reply:
> `ParallelMaster` creates the child jobs from **one** template job. It is by principle not possible to have different nodes.

This is not about the nodes, it is about the content of them.

Reply:
I don't understand. You raised the point that it takes only the nodes of the first job. The iteration over the jobs (which is anyway done) was never the issue here.